src/array.ts

```diff
@@ -396,3 +396,16 @@ export const diff = <T>(
   )
   return root.filter(a => !bKeys[identity(a)])
 }
+
+/**
+ * Shift array items by n steps
+ * If n > 0 items will shift n steps to the right
+ * If n < 0 items will shift n steps to the left
+ */
+ export function shift<T>(arr: Array<T>, n: number) {
+  if (n === 0 || arr.length === 0 || arr.length === n || n === -arr.length) return arr;
+
+  const shiftNumber = n % arr.length;
```

Comment:
Not totally convinced by it, since if I set n to a multiple of arr.length then it will do the computing operations
Here is a suggestion

```ts
  if (arr.length === 0) return arr;
  const shiftNumber = n % arr.length;
  if(shiftNumber === 0) return arr;
```

Reply:
This would match also multiples of `arr.length` (positive and negatice) as well as 0

Reply:
Then the implementation is even simpler and looks prettier:
```
  if (n === 0 || arr.length === 0) return arr;
  const shiftNumber = n % arr.length;
  if (shiftNumber === 0) return arr;
```

Reply:
Agree and upload

Reply:
Great! Now, simply add a test where you pass an empty array and the pipeline should not fail 😉